fdbserver/TLogServer.actor.cpp

```diff
@@ -2633,6 +2666,30 @@ ACTOR Future<Void> updateDurableClusterID(TLogData* self) {
 	}
 }
 
+ACTOR Future<Void> checkUpdateEncryptionAtRestMode(TLogData* self) {
+	EncryptionAtRestMode encryptionAtRestMode = wait(getEncryptionAtRestMode(self));
+
+	if (self->encryptionAtRestMode.present()) {
+		// Ensure the TLog encryptionAtRestMode status matches with the cluster config, if not, kill the TLog process.
+		// Approach prevents a fake TLog process joining the cluster.
+		TraceEvent("EncryptionAtRestMismatch", self->dbgid)
```

Comment:
This trace event name is misleading because it is logged even when there is not a mismatch